File: CP_Sheet_PH/Loops/N_Check_Code.cpp
/*
________________________
|                      |
|     AB Atiq          |
|______________________|
*/

// Problem link: https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/N
/*
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int a, b;
    cin >> a >> b;

    string digs;
    cin >> digs;

    if (digs[a] == '-')
    {
        bool chk = true;

        // Check first part
        for (int i = 0; i < a; i++)
        {
            if (digs[i] >= '0' && digs[i] <= '9')
            {
                continue;
            }
            else
            {
                chk = false;
                break;
            }
        }

        // Check second part
        for (int i = a + 1; i < a + b + 1; i++)
        {
            if (digs[i] >= '0' && digs[i] <= '9')
            {
                continue;
            }
            else
            {
                chk = false;
                break;
            }
        }

        if (chk)
            cout << "Yes";
        else
            cout << "No";
    }
    else
    {
        cout << "No";
    }

    return 0;
}
 */

#include <iostream>
using namespace std;

int main()
{
    int A, B;
    cin >> A >> B;

    string S;
    cin >> S;

    // Check if (A+1)-th position is '-'
    if (S[A] != '-')
    {
        cout << "No";
        return 0;
    }

    // Check all other characters are digits
    for (int i = 0; i < S.size(); i++)
    {
        if (i == A)
            continue; // skip '-'

        if (!(S[i] >= '0' && S[i] <= '9'))
        {
            cout << "No";
            return 0;
        }
    }

    cout << "Yes";
    return 0;
}

File: C-5-SDT/OOP_and_Python_Programming/M2.5/N_Check_Code.py
# https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/N
a,b = [int(x) for x in input().split()]
digs = input()
if digs[a] == '-':
    chk = True # check other digits without '-'
    for i in range(a):
        if digs[i] >= '0' and digs[i] <= '9':
            pass
        else:
            chk = False
            break
    for i in range(a+1, a+b+1):
        if digs[i] >= '0' and digs[i] <= '9':
            pass
        else:
            chk = False
            break
    if chk:
        print("Yes")
    else:        
        print("No")
        
else:
    print("No")